sinks/signalfx/signalfx.go

```diff
@@ -239,19 +239,26 @@ func MigrateConfig(conf *veneur.Config) error {
 	return nil
 }
 
-// CreateSignalFxSink creates a new SignalFx sink for metrics. This function
+// ParseConfig decodes the map config for a SignalFx sink into a
+// SignalFxSinkConfig struct.
+func ParseConfig(config interface{}) (interface{}, error) {
+	signalFxConfig := SignalFxSinkConfig{}
+	err := util.DecodeConfig(config, &signalFxConfig)
+	if err != nil {
+		return nil, err
+	}
+	return signalFxConfig, nil
+}
+
+// Create creates a new SignalFx sink for metrics. This function
 // should match the signature of a value in veneur.MetricSinkTypes, and is
 // intended to be passed into veneur.NewFromConfig to be called based on the
 // provided configuration.
-func CreateSignalFxSink(
+func Create(
 	server *veneur.Server, name string, logger *logrus.Entry,
 	config veneur.Config, sinkConfig interface{},
 ) (sinks.MetricSink, error) {
-	signalFxConfig := SignalFxSinkConfig{}
-	err := util.DecodeConfig(sinkConfig, &signalFxConfig)
-	if err != nil {
-		return nil, err
-	}
+	signalFxConfig := sinkConfig.(SignalFxSinkConfig)
```

Comment:
does this cast need to be guarded?

Reply:
Technically no, because we're passing the return value from `ParseConfig` into here, and the two should match. However, it's probably best to check just to be safe :) Done!

Reply:
I figured that was probably the case :) I don't mind whether or not we have the guard.